runtime/drivers/bigquery/sql_store.go

```diff
@@ -0,0 +1,220 @@
+	if unit == drivers.ProgressUnitRecord {
+		return r.bqIter.TotalRows, true
+	}
+
+	return 0, false
+}
+
+type row []any
+
+var _ bigquery.ValueLoader = &row{}
+
+func (r *row) Load(v []bigquery.Value, s bigquery.Schema) error {
+	m := make([]any, len(v))
+	for i := 0; i < len(v); i++ {
+		if s[i].Type == bigquery.RecordFieldType || s[i].Repeated {
+			return fmt.Errorf("repeated or nested data is not supported")
+		}
+
+		m[i] = convert(v[i])
+	}
+	*r = m
+	return nil
+}
+
+// type conversion table for time
+// bigquery(format) -- duckdb -- internal pb type
+// TIMESTAMP -- TIMESTAMPTZ/TIMESTAMP WITH TIME ZONE -- Type_CODE_TIMESTAMP
+// DATETIME(9999-12-31 23:59:59.999999) -- TIMESTAMP/DATETIME -- Type_CODE_DATETIME
+// TIME([H]H:[M]M:[S]S[.DDDDDD|.F]) -- TIME -- Type_CODE_TIME
+// DATE(YYYY-[M]M-[D]D) -- DATE -- Type_CODE_DATE
+func toPB(field *bigquery.FieldSchema) (*runtimev1.Type, error) {
+	t := &runtimev1.Type{Nullable: !field.Required}
+	switch field.Type {
+	case bigquery.StringFieldType:
+		t.Code = runtimev1.Type_CODE_STRING
+	case bigquery.JSONFieldType:
+		t.Code = runtimev1.Type_CODE_JSON
+	case bigquery.IntervalFieldType:
+		t.Code = runtimev1.Type_CODE_STRING
+	case bigquery.GeographyFieldType:
+		t.Code = runtimev1.Type_CODE_STRING
+	case bigquery.FloatFieldType:
+		t.Code = runtimev1.Type_CODE_FLOAT64
+	case bigquery.NumericFieldType:
+		t.Code = runtimev1.Type_CODE_DECIMAL
+	// big numeric can have width upto 76 digits which can't be converted to DECIMAL type in duckdb
+	// which supports width upto 38 digits only.
+	case bigquery.BigNumericFieldType:
+		t.Code = runtimev1.Type_CODE_STRING
+	case bigquery.TimestampFieldType:
+		t.Code = runtimev1.Type_CODE_TIMESTAMP
+	case bigquery.DateTimeFieldType:
+		t.Code = runtimev1.Type_CODE_DATETIME
+	case bigquery.TimeFieldType:
+		t.Code = runtimev1.Type_CODE_TIME
+	case bigquery.DateFieldType:
+		t.Code = runtimev1.Type_CODE_DATE
+	case bigquery.BooleanFieldType:
+		t.Code = runtimev1.Type_CODE_BOOL
+	case bigquery.IntegerFieldType:
+		t.Code = runtimev1.Type_CODE_INT64
+	case bigquery.BytesFieldType:
+		t.Code = runtimev1.Type_CODE_BYTES
+	case bigquery.RecordFieldType:
+		return nil, fmt.Errorf("record type not supported")
+	default:
+		// TODO :: may be just use VARCHAR ?
+		return nil, fmt.Errorf("type %s not supported", field.Type)
+	}
+	return t, nil
+}
+
+func convert(v any) any {
+	if v == nil {
+		return nil
+	}
+	// refer to documentation on bigquery.RowIterator.Next for the superset of all go types possible
+	switch val := v.(type) {
+	case civil.Date:
+		return val.In(time.UTC)
+	case civil.Time:
+		t, _ := time.Parse("15:04:05.999999999", val.String())
+		return t
+	case civil.DateTime:
+		return val.In(time.UTC)
+	case *big.Rat:
+		f, _ := val.Float64()
+		if math.IsInf(f, 0) {
+			// big.Rat can't always be represented in float64
+			// we use string notation(in the form num/denom) in such cases which is not ideal
+			return val.String()
+		}
+		return f
```

Comment:
1. What do you think of just always returning a string for big numbers? So people get predictable behavior that BigQuery NUMERIC always becomes DuckDB TEXT?
2. Is there any way to make the big rat return a decimal string instead of a fraction string? Given this is coming from BigQuery NUMERIC/DECIMAL, I guess we know it's never a fraction with an infinite decimal representation (though should be truncated to be safe)?

Reply:
Yeah it should be always possible to represent numeric in `varchar` and I also looked into the `big.Rat` package and it should be possible to represent it in decimal string as well. 
The only issue is that user experience will take a hit in since it will always be required to cast it back to double/decimal. We will also not show min/max/avg value right ?
I think we can take this approach and change it later based on user feedback ? We may even think of adding `big.Rat` support directly into duckdb driver.